samples/cpp/common/utils/include/samples/common.hpp

```diff
@@ -436,10 +436,10 @@ static UNUSED void addRectangles(unsigned char* data,
             thickness = 1;
         }
 
-        if (static_cast<std::size_t>(x + w) >= width) {
+        if ((static_cast<std::size_t>(x) + w) >= width) {
```

Comment:
how overflow cam happen?

Reply:
`x` and `w` are of `int32_t` type and their sum is of `int32_t` type, so if the sum doesn't fit `int32_t` range, it will be overflowed and, then, casted to `std::size_t`.
But when one operand is casted to `std::size_t`, so the another one will be promoted automatically.